jolie/src/main/java/jolie/OOITBuilder.java

```diff
@@ -565,48 +565,44 @@ public void visit( InputPortInfo n ) {
 		locationPath = new ClosedVariablePath( locationPath, interpreter.globalValue() );
 
 		Expression locationExpr = buildExpression( n.location() );
-		Value locationVal = null;
-		String location = null;
+		String locationStr = null;
 		if( locationExpr instanceof Value ) {
-			locationVal = locationExpr.evaluate();
+			locationStr = locationExpr.evaluate().strValue();
 		}
 
-		if( locationVal == null ) {
+		if( locationStr == null ) {
 			error( n.context(), "location expression is not valid" );
 			return;
-		} else {
-			location = locationVal.strValue();
-			locationPath.getValue().setValue( location );
 		}
+		locationPath.getValue().setValue( locationStr );
 
 
-		Expression protocolExpr = null;
-		String protocol = null;
+		String protocolStr = null;
+		List< Process > protocolProcs = new ArrayList<>();
 		if( n.protocol() != null ) {
-			OLSyntaxNode protocolNode = ModuleSystemUtil.transformProtocolExpression( n.protocol() );
-			Value protocolVal = null;
-			protocolExpr = buildExpression( protocolNode );
-			if( protocolExpr instanceof Value || protocolExpr instanceof InlineTreeExpression ) {
-				protocolVal = protocolExpr.evaluate();
-			}
-
-			if( protocolVal == null ) {
+			OLSyntaxNode protocolIdNode =
+				ModuleSystemUtil.transformProtocolId( n.protocolId(), interpreter.commCore() );
+			Expression protocolExpr = buildExpression( n.protocol() );
+			Expression protocolIdExpr = buildExpression( protocolIdNode );
+			protocolProcs.add( new DeepCopyProcess( protocolPath, protocolExpr, true, n.context() ) );
+			protocolProcs.add( new AssignmentProcess( protocolPath, protocolIdExpr, n.context() ) );
```

Comment:
Here, we use ModuleSystemUtil.transformProtocolId to decide whether if we should transform a VariablePathExpressionNode which defines protocolId, into ConstantString or not, by determine if the identifier is supported by Jolie or not.